### ISSUE TITLE
ModernHttpClient crashes on Android when used in SignalR

### ISSUE THREAD

**Issue by peterno** [paulcbetts/ModernHttpClient#136]
Using ModernHttpClient and SignalR together gives great benefits related to performance etc and also having the connection reconnecting when switching from wifi to 3g/4g.

To use ModernHttpClient together with SignalR my solution was to subclass the DefaultHttpClient from SignalR and use that instance as input to the Start method.
```C#
public class ModernHttpClientSignalR : DefaultHttpClient
{
        protected override HttpMessageHandler CreateHandler()
        {
            return new NativeMessageHandler();
        }
}

await _connection.Start(new ModernHttpClientSignalR());
```
This works nicely on iOS.
 On Android the ModernHttpClient does not like it when SignalR is posting data without any ContentType and I had to fix line 83 in OkHttpNetworkHandler.cs.

From: 
```C#
var contentType = String.Join (" ", request.Content.Headers.GetValues ("Content-Type"));
```
To:
```C#
var contentType = string.Empty;
if (request.Content.Headers.ContentType != null)
{
    contentType = String.Join(" ", request.Content.Headers.GetValues("Content-Type"));
}
```
It would be great if this fix could be part of the original code.

**Comment by paulcbetts** [comment]
Can you send a pull request to this effect?

**Comment by peterno** [comment]
Hi Paul,
I am not sure what you mean? Do you want a pull request as an example when the error occurs or do you want a pull request with the fix? I am a little new to the git way of working:-)

**Comment by peterno** [comment]
I have added pull request #141.

**Comment by paulcbetts** [comment]
Fixed in #141
